File: Projects/Week 1/cash.c
#include <cs50.h>
#include <stdio.h>

void factor(int n)
{
    int r1, r2, r3, r4;
    int l1, l2, l3, l4;
    l1 = n / 25;
    r1 = n % 25;
    l2 = r1 / 10;
    r2 = r1 % 10;
    l3 = r2 / 5;
    r3 = r2 % 5;
    l4 = r3 / 1;
    r4 = r3 % 1;
    int total = l1 + l2 + l3 + l4;
    printf("%d\n", total);
}

int main()
{
    int n;
    do
    {
        n = get_int("Change owed: ");
    }
    while (n < 1);
    {
        factor(n);
    }
}

File: CS50x/Problem Sets/Week 6/cash.py
from cs50 import get_float


def factor(n):
    l1 = n // 25
    r1 = n % 25
    l2 = r1 // 10
    r2 = r1 % 10
    l3 = r2 // 5
    r3 = r2 % 5
    l4 = r3 // 1
    r4 = r3 % 1
    total = l1 + l2 + l3 + l4
    p = int(total)
    print(f"{p}")


while True:
    cents = get_float("Change: ")
    if cents > 0:
        break
cents = round(cents * 100)
factor(cents)

# while cents >= 25:
#     cents = cents - 25
#     count += 1
# while cents >= 10:
#     cents = cents - 10
#     count += 1
# while cents >= 5:
#     cents = cents - 5
#     count += 1

# while cents >= 1:
#     cents = cents - 1
#     count += 1
